static-instrumenter/maven-plugin/build.gradle.kts

```diff
@@ -1,5 +1,32 @@
 plugins {
   id("otel.java-conventions")
+  id("otel.publish-conventions")
 }
 
 description = "Maven3 plugin for static instrumentation of projects code and dependencies"
+extra["mavenArtifactId"] = "static-instrumentation-maven-plugin"
```

Comment:
Is there a reason we need this instead of setting `baseArchives.name`?

Reply:
The original ask from @mateuszrzeszutek was to publish the plugin under a specific artifactId - "static-instrumentation-maven-plugin". So this is the way I came up with, mirroring the solution for groupId we had.
As my knowledge of gradle is quite limited, it's probably a question to you - is it sufficient to change the baseArchives.name in the project? 😄

Reply:
Done, looks like it's working.